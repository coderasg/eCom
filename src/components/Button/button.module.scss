@use "./../../variables";

.button {
  padding: variables.$paddingMid;
  border: none;
  background-color: variables.$checkoutButton;
  color: variables.$white;
}
