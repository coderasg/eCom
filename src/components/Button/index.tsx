import React, { MouseEvent, ReactNode } from 'react';


import styles from './button.module.scss';

type Props = {
  className: string;
  label?: string;
  children?: ReactNode;
  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
};

function Button(props: Props) {
  const { label, className, children, onClick } = props;

  return (
    <button data-testid="Button-wrapper" onClick={onClick} className={`${styles.button} ${className}`}>
      {label}
      {children}
    </button>
  );
}

export default Button;
