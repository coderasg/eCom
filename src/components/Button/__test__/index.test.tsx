import React, {MouseEvent} from 'react';
import renderer from 'react-test-renderer';

import Button from '../index';

describe('(Component) Button', () => {
  const defaultProps = {
    className: 'className',
    onClick: (event: MouseEvent<HTMLButtonElement>) => {
      console.log('click', event);
    },
  };

  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<Button {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
