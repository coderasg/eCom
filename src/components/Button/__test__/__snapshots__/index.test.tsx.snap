// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Button should pass snapshot test 1`] = `
<button
  className="undefined className"
  data-testid="Button-wrapper"
  onClick={[Function]}
/>
`;
